README.md

```diff
@@ -311,6 +289,140 @@ ICP based refinement can be applied to the poses of the input pointclouds before
+      <param name="esdf_max_distance_m" value="2.0" />
+      <param name="color_mode" value="color" />
+      <param name="use_tf_transforms" value="true" />
+      <param name="update_mesh_every_n_sec" value="1.0" />
+      <param name="clear_sphere_for_planning" value="true" />
+      <param name="slice_level" value="1.0" />
+      <param name="world_frame" value="$(arg world_frame)" />
+      <param name="verbose" value="false" />
+    </node>
+
+    <node name="my_voxblox_planner" pkg="voxblox_planner" type="my_voxblox_planner" output="screen" args="-alsologtostderr">
+      <remap from="odometry" to="great_estimator/odometry" />
+      <remap from="my_voxblox_planner/esdf_map_in" to="esdf_map" />
+      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
+      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
+      <param name="update_mesh_every_n_sec" value="0.0" />
+      <param name="world_frame" value="$(arg world_frame)" />
+      <param name="verbose" value="false" />
+    </node>
+
+  </group>
+</launch>
+
+```
+
+And some scaffolding for writing your own planner using ESDF collision checking:
+```c++
+class YourPlannerVoxblox {
+ public:
+  YourPlannerVoxblox(const ros::NodeHandle& nh,
+                    const ros::NodeHandle& nh_private);
+  virtual ~YourPlannerVoxblox() {}
+  double getMapDistance(const Eigen::Vector3d& position) const;
+ private:
+  ros::NodeHandle nh_;
+  ros::NodeHandle nh_private_;
+
+  // Map!
+  voxblox::EsdfServer voxblox_server_;
+};
+```
+There's also a traversability pointcloud you can enable/disable, that if you set the radius to your robot's collision checking radius, can show you parts of the map the planner thinks are traversable in a pointcloud:
+```c++
+YourPlannerVoxblox::YourPlannerVoxblox(const ros::NodeHandle& nh,
+                                     const ros::NodeHandle& nh_private)
+    : nh_(nh),
+      nh_private_(nh_private),
+      voxblox_server_(nh_, nh_private_) {
+  // Optionally load a map saved with the save_map service call in voxblox.
+  std::string input_filepath;
+  nh_private_.param("voxblox_path", input_filepath, input_filepath);
+  if (!input_filepath.empty()) {
+    if (!voxblox_server_.loadMap(input_filepath)) {
+      ROS_ERROR("Couldn't load ESDF map!");
+    }
+  }
+  double robot_radius = 1.0;
+  voxblox_server_.setTraversabilityRadius(robot_radius);
+  voxblox_server_.publishTraversable();
+}
+```
+
+Then to check for collisions you can just compare map distance to your robot radius:
+```c++
+double YourPlannerVoxblox::getMapDistance(
+    const Eigen::Vector3d& position) const {
+  if (!voxblox_server_.getEsdfMapPtr()) {
+    return 0.0;
+  }
+  double distance = 0.0;
+  if (!voxblox_server_.getEsdfMapPtr()->getDistanceAtPosition(position,
+                                                              &distance)) {
+    return 0.0;
+  }
+  return distance;
+}
+```
+
+# Transformations in Voxblox
```

Comment:
our readme is getting really long, at some point maybe we should turn it into a wiki and just have short summaries and links in it

Reply:
Yeah I agree. Too lazy to do it today though. ;) I think next section we add it's unavoidable.